### PR TITLE
fix: mask with valid data when preCheckOverflow

### PR DESCRIPTION
**Pull request by smellthemoon**
#37175

### CODE REVIEW

File: internal/core/src/exec/expression/UnaryExpr.cpp [hunk 1]
```diff
@@ -758,53 +758,38 @@ PhyUnaryRangeFilterExpr::PreCheckOverflow() {
                 cached_overflow_res_->size() == batch_size) {
                 return cached_overflow_res_;
             }
+            auto valid = ProcessChunksForValid<T>(CanUseIndex<T>());
+            auto res_vec = std::make_shared<ColumnVector>(
+                TargetBitmap(batch_size), std::move(valid));
+            TargetBitmapView res(res_vec->GetRawData(), batch_size);
+            TargetBitmapView valid_res(res_vec->GetValidRawData(), batch_size);
+            cached_overflow_res_ = res_vec;
```
Comment (chasingegg) [inline internal/core/src/exec/expression/UnaryExpr.cpp:766]: seems cached_overflow_res_ no longer useful, bcz valid_data may be different in different batches

**Comment by smellthemoon** [comment]
rerun cpp-unit-test

**Comment by smellthemoon** [comment]
rerun go-sdk

**Comment by smellthemoon** [comment]
rerun go-sdk

**Review by congqixia — APPROVED** [review]
/approve

**Pull request merged**